### PR TITLE
Feat: Support function name list in ->

### PR DESCRIPTION
**Pull request by kilianmh**
- Add support for multiple function names when declaring types with `->`
- Add unit tests for one and multiple functions names

### CODE REVIEW

**Comment by ruricolist** [comment]
There's a deeper issue here: this would mean it's no longer possible to use `->` to declare ftypes for setf functions.

**Comment by ruricolist** [comment]
One approach might be to dispatch on whether the first element of the list is a symbol in the :cl package; if so treat it as a function name that's a list; otherwise treat it as a list of function names.

**Comment by kilianmh** [comment]
Sure we can test whether its part of the `cl` package. Alternatively, if `setf` is the only special case, we could check if the first element is the `setf` symbol.

**Comment by kilianmh** [comment]
I added now the check whether the first symbol is in the `cl` package. Now `setf` should work. Also there are more test-cases. 
Anything else needs to change?

File: tests/types.lisp [hunk 1]
```diff
@@ -17,6 +17,19 @@
...
+    (let ((function-name-1 (gensym))
+	  (function-name-2 (gensym))
+	  (function-name-3 (gensym))
+	  (function-name-4 (gensym))
+	  (function-name-5 (gensym))
+	  (function-name-6 (gensym)))
+      `(-> ,function-name-1 (integer) integer)
+      `(-> (,function-name-2 ,function-name-3) (integer) integer)
+      `(-> (setf ,function-name-4) (real) real)
+      `(-> (,function-name-5 (setf ,function-name-6)) (real) real))))
```
Comment (ruricolist) [inline tests/types.lisp:31]: What is this actually testing (besides creating lists)? Did you mean to macroexpand these?

Comment (kilianmh) [reply tests/types.lisp]: These tests were silly I have to admit. Now they use `macroexpand-1` and `equalp` to compare. We might also use `equal` for comparison if you want that.

**Comment by ruricolist** [comment]
Merged, thanks!

**Pull request merged**